Combobox.js

```diff
@@ -641,65 +665,145 @@ define([
+				this.inputNode.value = selectedItem ? this._getItemLabel(selectedItem) : "";
+				this.value = selectedItem ? this._getItemValue(selectedItem) : "";
+			} else {
+				this.inputNode.value = this.displayedValue !== "" ? this.displayedValue : this.value;
 			}
-			this._set("value", value);
-			// FormWidget.refreshRendering() also updates valueNode.value, but we need to
-			// make sure this is already done when FormValueWidget.handleOnInput() runs.
-			this.valueNode.value = value;
-			this.handleOnInput(this.value); // emit "input" event
+		},
+
+		_validateMultiple: function (userInteraction) {
+			var n;
+			if (userInteraction) {
+				var selectedItems = this.list.selectedItems;
+				var inputElement = this._popupInput || this.inputNode;
+				n = selectedItems ? selectedItems.length : 0;
+				var value = [];
+				if (n > 1) {
+					inputElement.value = this.multipleChoiceMsg;
+					for (var i = 0; i < n; i++) {
+						value.push(selectedItems[i] ? this._getItemValue(selectedItems[i]) : "");
+					}
+				} else if (n === 1) {
+					var selectedItem = this.list.selectedItem;
+					inputElement.value = this._getItemLabel(selectedItem);
+					value.push(this._getItemValue(selectedItem));
+				} else { // no option selected
+					inputElement.value = this.multipleChoiceNoSelectionMsg;
+				}
+				this._set("value", value);
+				// FormWidget.refreshRendering() also updates valueNode.value, but we need to
+				// make sure this is already done when FormValueWidget.handleOnInput() runs.
+				this.valueNode.value = value;
+				this.handleOnInput(this.value); // emit "input" event
+			} else {
+				var items = [];
+				if (typeof this.value === "string" && this.value.length > 0) {
+					items = this.value = this.value.split(",");
+					this.valueNode.value = this.value;
+				} else if (this.value instanceof Array) {
+					items = this.value;
+				} // else empty array. No pre-set values.
+				n = items.length;
+				if (n > 1) {
+					this.inputNode.value = this.multipleChoiceMsg;
+				} else if (n === 1) {
+					this.inputNode.value = this.displayedValue !== "" ? this.displayedValue : items[0];
+				} else {
+					this.inputNode.value = this.multipleChoiceNoSelectionMsg;
+				}
+			}
+
 		},
 
 		/**
 		 * Filters the embedded List to only show the items matching `filterTxt`.
 		 * If `autoFilter` is `true` and `selectionMode` is `"single"`, the method
 		 * is called automatically while the user types into the editable input
 		 * element, with `filterTxt` being the currently entered text.
-		 * The default implementation uses `dstore/Filter.match()`.
-		 * The matching is performed against the `list.labelAttr` attribute of
-		 * the data store items.
-		 * The method can be overridden for implementing other filtering strategies.
 		 * @protected
 		 */
-		filter: function (filterTxt) {
-			if (this.filterMode === "startsWith") {
+		filter: function (inputText) {
+			// Escape special chars in search string, see
```

Comment:
This line:

```
// Escape special chars in search string, see
```

is indented more than:

```
// http://stackoverflow.com/questions/3446170/escape-string-for-use-in-javascript-regex.
```

and subsequent lines.  It might also be a problem due to using spaces instead of tabs.